test/common/testlib.py

```diff
@@ -230,15 +230,21 @@ def blur(self, selector):
 
     def key_press(self, keys):
         for k in keys:
-            if k.isalnum():
+            if k == "Backspace":
+                self.cdp.invoke("Input.dispatchKeyEvent", type="keyDown", windowsVirtualKeyCode=8)
+            elif k.isalnum():
                 self.cdp.invoke("Input.dispatchKeyEvent", type="char", text=k, key=k)
             else:
                 self.cdp.invoke("Input.dispatchKeyEvent", type="char", text=k)
 
     def set_input_text(self, selector, val):
         self.set_val(selector, "")
         self.focus(selector)
-        self.key_press(val)
+        if val == "":
+            # We need some real action for React to emit change signals
+            self.key_press([ " ", "Backspace" ])
```

Comment:
Will that actually be enough if the field already has some value? But I figure that would have been a problem before already. Perhaps `focus()` has the same behaviour as tabbing to an element, i. e. it marks the entire content and the space then overwrites it? But in that case, just sending a backspace ought to be enough?

Reply:
> Will that actually be enough if the field already has some value?

Yes, because of the `set_val(selector, "")` at the beginning.  But this also feels too magical.  If setting the value works, why do we need to simulate key strokes?  Probably only to cause React to send change notifications.  With the "C-a DEL <text>" sequence we could get rid of that piece of magic as well.  Or we do set_val(text), and then always  "SPC Backspace" just to send the notification.  Or we figure out how this is supposed to be done by asking React peeps.

Reply:
Right, I missed that initial `set_val`, that's also what makes that actually work in master.